src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -208,20 +217,74 @@ public WindowsIdentity(string sUserPrincipalName)
 
                             if (profileBuffer != null)
                                 profileBuffer.Dispose();
-
-                            _safeTokenHandle = accessTokenHandle;
                         }
                     }
                 }
             }
         }
 
-        private static SafeLsaHandle ConnectToLsa()
+        /// <summary>
+        /// To obtain SafeLsaHandle we try before for an Impersonation token that require SeTcbPrivilege, if fails we fallback
+        /// to an Identification token
+        /// </summary>
+        /// <param name="logonProcessNameBuffer"></param>
+        /// <returns></returns>
+        private static SafeLsaHandle GetLsaHandle(byte[] logonProcessNameBuffer)
         {
             SafeLsaHandle lsaHandle;
-            int ntStatus = Interop.SspiCli.LsaConnectUntrusted(out lsaHandle);
-            if (ntStatus < 0) // non-negative numbers indicate success
-                throw GetExceptionFromNtStatus(ntStatus);
+
+            // Try to get an impersonation token
+            Privilege privilege = null;
+            try
+            { 
+                // Try to enable the TCB privilege if possible
+                try
+                {
+                    privilege = new Privilege("SeTcbPrivilege");
+                    privilege.Enable();
+                }
+                catch (PrivilegeNotHeldException) { }
```

Comment:
If Enable throws, we'll still end up calling Revert later on.  Is that ok?  Presumably "yes", because that's what netfx does.

Reply:
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Security.AccessControl/src/System/Security/AccessControl/Privilege.cs#L518 
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Security.AccessControl/src/System/Security/AccessControl/Privilege.cs#L561

Revert will be skipped if state doesn't change.